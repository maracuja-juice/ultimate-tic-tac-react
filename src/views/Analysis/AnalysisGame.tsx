import * as React from 'react';
import { BigBoard } from '../../components/Board/BigBoard/BigBoard';
import { RouteComponentProps } from 'react-router';
import { connect } from 'react-redux';
import {
  AppState,
  Move,
  Player,
  SmallBoardInformation,
} from '../../state/AppState';
import {
  loadFinishedGame,
  moveBackwardInHistory,
  moveForwardInHistory,
} from '../../state/analysisGame/analysisGameActions';
import './AnalysisGame.css';
import { Point } from '../../util/Point';
import { HistoryButtons } from '../../components/Analysis/HistoryButtons/HistoryButtons';
import { MoveList } from '../../components/Analysis/MoveList/MoveList';
import { scroller } from 'react-scroll/modules';
import { moveScrollElementBaseName } from '../../components/Analysis/ScrollElementConstants';

interface AnalysisGameProps extends RouteComponentProps<{ id: string }> {
  onLoad: (id: string) => void;
  moveForwardInHistory: (numberOfMoves: number) => void;
  moveBackwardInHistory: (numberOfMoves: number) => void;
  reversedMoves?: Move[];
  board: SmallBoardInformation[];
  activeBoards: Point[];
  currentPlayer: Player;
  currentMove: number;
}

export class AnalysisGame extends React.Component<AnalysisGameProps> {
  componentDidMount = () => {
    const id = this.props.match.params.id;
    this.props.onLoad(id);
  };

  scrollToElement = (moveNumberToScrollTo: number) => {
    scroller.scrollTo(moveScrollElementBaseName + moveNumberToScrollTo, {
      duration: 300,
      smooth: true,
      containerId: 'moveList',
      offset: -108,
    });
  };

  render() {
    const {
      reversedMoves,
      board,
      activeBoards,
      currentPlayer,
      currentMove,
    } = this.props;

    const currentlyAppliedMove =
      reversedMoves &&
      reversedMoves[
        reversedMoves.findIndex((m) => m.moveNumber === currentMove)!
      ];

    return (
      <div className="center">
        <div className="analysisLayout">
          <div id="moveList" className="moveList">
            {reversedMoves && (
              <MoveList
                currentMove={currentMove}
                reversedMoves={reversedMoves}
                moveForwardInHistory={this.props.moveForwardInHistory}
                moveBackwardInHistory={this.props.moveBackwardInHistory}
              />
            )}
          </div>
          <div className="historyButtons">
            {reversedMoves && (
              <HistoryButtons
                currentMove={currentMove}
                lastMove={reversedMoves[0] && reversedMoves[0].moveNumber}
                moveForwardInHistory={this.props.moveForwardInHistory}
                moveBackwardInHistory={this.props.moveBackwardInHistory}
                onInteraction={this.scrollToElement}
              />
            )}
          </div>
          <div className="analysisGame">
            {reversedMoves && (
              <BigBoard
                // tslint:disable-next-line:no-empty
                onPlayerMoved={() => {}}
                board={board}
                activeBoards={activeBoards}
                currentPlayer={currentPlayer}
                markTileSpecially={{
                  condition: currentlyAppliedMove !== undefined,
                  position: currentlyAppliedMove
                    ? {
                        boardPosition: currentlyAppliedMove.boardPosition,
                        tilePosition: currentlyAppliedMove.tilePosition,
                      }
                    : undefined,
                }}
              />
            )}
          </div>
        </div>
      </div>
    );
  }
}

const mapStateToProps = (state: AppState) => ({
  reversedMoves: state.analysisGame.moves.slice().reverse(),
  board: state.analysisGame.board,
  activeBoards: state.analysisGame.activeBoards,
  currentPlayer: state.analysisGame.game.currentPlayer,
  currentMove: state.analysisGame.currentMove,
});

// tslint:disable-next-line: no-any
const mapDispatchToProps = (dispatch: any) => ({
  onLoad: (id: string) => dispatch(loadFinishedGame(id)),
  moveForwardInHistory: (numberOfMoves: number) =>
    dispatch(moveForwardInHistory(numberOfMoves)),
  moveBackwardInHistory: (numberOfMoves: number) =>
    dispatch(moveBackwardInHistory(numberOfMoves)),
});

export default connect(
  mapStateToProps,
  mapDispatchToProps,
)(AnalysisGame);
