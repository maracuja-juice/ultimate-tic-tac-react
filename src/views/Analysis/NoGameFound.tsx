import * as React from 'react';
import { Typography } from '@rmwc/typography';
import styles from './NoGameFound.module.css';
import classNames from 'classnames';

interface NoGameFoundProps {
  center?: boolean;
  className?: string;
  tag: string;
}

export function NoGameFound(props: NoGameFoundProps) {
  return (
    <div
      className={classNames(props.className, {
        [styles.noGameFoundLayout]: props.center,
      })}
    >
      <Typography use="headline3" tag={props.tag}>
        No game was found
      </Typography>
    </div>
  );
}
