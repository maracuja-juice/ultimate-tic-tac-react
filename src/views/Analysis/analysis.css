.mdc-list {
    padding: 0 !important;
}

.analysisLayout {
    display: grid;
    grid-gap: 1em;
    grid-template-areas: "moveList" "historyButtons" "game";
    height: 100%;
}

.moveList {
    grid-area: moveList;
    padding: 0;
    overflow-y: scroll;
    min-height: 50vmin;
}

.analysisGame {
    grid-area: game;
}

.historyButtons {
    grid-area: historyButtons;
    display: flex;
    justify-content: center;
    max-height: 10vmin;
}

.historyButtons button {
    margin: 5px;
}

@media only screen and (min-width: 641px) and (max-height: 768px), only screen and (min-width: 1350px)  {
    .moveList {
        max-height: 81vmin;
    }

    .analysisLayout {
        grid-template-columns: minmax(20vmin, 1fr) auto;
        grid-template-rows: minmax(9vmin, 12vmin) 81vmin;
        grid-template-areas: ". historyButtons" "moveList game";
        align-items: center;
        grid-column-gap: 1.5em;
    }

    .analysisLayout .mdc-list-item__graphic {
        margin-right: 17px !important;
    }

}