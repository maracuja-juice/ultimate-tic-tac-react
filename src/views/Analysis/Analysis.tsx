import * as React from 'react';
import { BigBoard } from '../../components/Board/BigBoard/BigBoard';
import { RouteComponentProps } from 'react-router';
import { connect } from 'react-redux';
import {
  AppState,
  Move,
  Player,
  SmallBoardInformation,
} from '../../state/AppState';
import {
  loadFinishedGame,
  moveBackwardInHistory,
  moveForwardInHistory,
} from '../../state/analysisGame/analysisGameActions';
import './analysis.css';
import { Point } from '../../util/Point';
import { HistoryButtons } from '../../components/Analysis/HistoryButtons';
import { MoveList } from '../../components/Analysis/MoveList';

interface AnalysisProps extends RouteComponentProps<{ id: string }> {
  onLoad: (id: string) => void;
  moveForwardInHistory: (numberOfMoves: number) => void;
  moveBackwardInHistory: (numberOfMoves: number) => void;
  reversedMoves?: Move[];
  board: SmallBoardInformation[];
  activeBoards: Point[];
  currentPlayer: Player;
  currentMove: number;
}

export class Analysis extends React.Component<AnalysisProps> {
  componentDidMount = () => {
    const id = this.props.match.params.id;
    this.props.onLoad(id);
  };

  render() {
    const {
      reversedMoves,
      board,
      activeBoards,
      currentPlayer,
      currentMove,
    } = this.props;

    const currentlyAppliedMove =
      reversedMoves &&
      reversedMoves[
        reversedMoves.findIndex((m) => m.moveNumber === currentMove)!
      ];

    return (
      <div className="center">
        <div className="analysisLayout">
          <div id="moveList" className="moveList">
            {reversedMoves && (
              <MoveList
                currentMove={currentMove}
                reversedMoves={reversedMoves}
                moveForwardInHistory={this.props.moveForwardInHistory}
                moveBackwardInHistory={this.props.moveBackwardInHistory}
              />
            )}
          </div>
          <div className="historyButtons">
            {reversedMoves && (
              <HistoryButtons
                currentMove={currentMove}
                lastMove={reversedMoves[0] && reversedMoves[0].moveNumber}
                moveForwardInHistory={this.props.moveForwardInHistory}
                moveBackwardInHistory={this.props.moveBackwardInHistory}
              />
            )}
          </div>
          <div className="analysisGame">
            {reversedMoves && (
              <BigBoard
                // tslint:disable-next-line:no-empty
                onPlayerMoved={() => {}}
                board={board}
                activeBoards={activeBoards}
                currentPlayer={currentPlayer}
                markTileSpecially={{
                  condition: currentlyAppliedMove !== undefined,
                  position: currentlyAppliedMove
                    ? {
                        boardPosition: currentlyAppliedMove.boardPosition,
                        tilePosition: currentlyAppliedMove.tilePosition,
                      }
                    : undefined,
                }}
              />
            )}
          </div>
        </div>
      </div>
    );
  }
}

const mapStateToProps = (state: AppState) => ({
  reversedMoves: state.analysisGame.moves.slice().reverse(),
  board: state.analysisGame.board,
  activeBoards: state.analysisGame.activeBoards,
  currentPlayer: state.analysisGame.game.currentPlayer,
  currentMove: state.analysisGame.currentMove,
});

// tslint:disable-next-line: no-any
const mapDispatchToProps = (dispatch: any) => ({
  onLoad: (id: string) => dispatch(loadFinishedGame(id)),
  moveForwardInHistory: (numberOfMoves: number) =>
    dispatch(moveForwardInHistory(numberOfMoves)),
  moveBackwardInHistory: (numberOfMoves: number) =>
    dispatch(moveBackwardInHistory(numberOfMoves)),
});

export default connect(
  mapStateToProps,
  mapDispatchToProps,
)(Analysis);
