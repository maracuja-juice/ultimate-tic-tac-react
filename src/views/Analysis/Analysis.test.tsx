import * as React from 'react';
import { configure, shallow } from 'enzyme';
import * as ReactSixteenAdapter from 'enzyme-adapter-react-16';
import { Analysis } from './Analysis';
import { circleFinishedBoardMock } from '../../__mocks__/finishedBoardMock';
import { Player } from '../../state/AppState';

configure( {adapter: new ReactSixteenAdapter()} );

describe('Analysis', () => {
    it('should match snapshot', () => {
        // tslint:disable:no-empty
        const onLoad = jest.fn((id) => {});
        const moveForwardInHistory = jest.fn((number) => {});
        const moveBackwardInHistory = jest.fn((number) => {});
        const mock: any = jest.fn();
        const match = {
            isExact: true,
            params: {
                id: 'someId'
            },
            path: "",
            url: ""
        }

        const analysis = shallow(
           <Analysis
               onLoad={onLoad}
               moveForwardInHistory={moveForwardInHistory}
               moveBackwardInHistory={moveBackwardInHistory}
               board={circleFinishedBoardMock}
               activeBoards={[]}
               currentPlayer={Player.Cross}
               currentMove={1}
               history={mock}
               location={mock}
               match={match}
           />
       );

        expect(analysis).toMatchSnapshot();
    });

    it('should call onLoad with givenId when mounted', () => {
        // tslint:disable:no-empty
        const onLoad = jest.fn((id) => {});
        const moveForwardInHistory = jest.fn((number) => {});
        const moveBackwardInHistory = jest.fn((number) => {});
        const mock: any = jest.fn();
        const match = {
            isExact: true,
            params: {
                id: '123'
            },
            path: "",
            url: ""
        }

        shallow(
            <Analysis
                onLoad={onLoad}
                moveForwardInHistory={moveForwardInHistory}
                moveBackwardInHistory={moveBackwardInHistory}
                board={circleFinishedBoardMock}
                activeBoards={[]}
                currentPlayer={Player.Cross}
                currentMove={1}
                history={mock}
                location={mock}
                match={match}
            />
        );

        expect(onLoad).toHaveBeenCalledWith('123');
    });
});