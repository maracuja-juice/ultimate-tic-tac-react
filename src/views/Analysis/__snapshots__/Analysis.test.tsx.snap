// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Analysis should match snapshot 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Analysis
    activeBoards={Array []}
    board={
        Array [
            Object {
              "position": Object {
                "x": 0,
                "y": 0,
              },
              "tiles": Array [
                Object {
                  "boardPosition": Object {
                    "x": 0,
                    "y": 0,
                  },
                  "position": Object {
                    "x": 0,
                    "y": 0,
                  },
                  "value": 0,
                },
                Object {
                  "boardPosition": Object {
                    "x": 0,
                    "y": 0,
                  },
                  "position": Object {
                    "x": 0,
                    "y": 1,
                  },
                  "value": 1,
                },
                Object {
                  "boardPosition": Object {
                    "x": 0,
                    "y": 0,
                  },
                  "position": Object {
                    "x": 0,
                    "y": 2,
                  },
                  "value": 2,
                },
                Object {
                  "boardPosition": Object {
                    "x": 0,
                    "y": 0,
                  },
                  "position": Object {
                    "x": 1,
                    "y": 0,
                  },
                  "value": 2,
                },
                Object {
                  "boardPosition": Object {
                    "x": 0,
                    "y": 0,
                  },
                  "position": Object {
                    "x": 1,
                    "y": 1,
                  },
                  "value": 1,
                },
                Object {
                  "boardPosition": Object {
                    "x": 0,
                    "y": 0,
                  },
                  "position": Object {
                    "x": 1,
                    "y": 2,
                  },
                  "value": 2,
                },
                Object {
                  "boardPosition": Object {
                    "x": 0,
                    "y": 0,
                  },
                  "position": Object {
                    "x": 2,
                    "y": 0,
                  },
                  "value": 2,
                },
                Object {
                  "boardPosition": Object {
                    "x": 0,
                    "y": 0,
                  },
                  "position": Object {
                    "x": 2,
                    "y": 1,
                  },
                  "value": 1,
                },
                Object {
                  "boardPosition": Object {
                    "x": 0,
                    "y": 0,
                  },
                  "position": Object {
                    "x": 2,
                    "y": 2,
                  },
                  "value": 2,
                },
              ],
              "value": 1,
            },
            Object {
              "position": Object {
                "x": 0,
                "y": 1,
              },
              "tiles": Array [
                Object {
                  "boardPosition": Object {
                    "x": 0,
                    "y": 1,
                  },
                  "position": Object {
                    "x": 0,
                    "y": 0,
                  },
                  "value": 0,
                },
                Object {
                  "boardPosition": Object {
                    "x": 0,
                    "y": 1,
                  },
                  "position": Object {
                    "x": 0,
                    "y": 1,
                  },
                  "value": 1,
                },
                Object {
                  "boardPosition": Object {
                    "x": 0,
                    "y": 1,
                  },
                  "position": Object {
                    "x": 0,
                    "y": 2,
                  },
                  "value": 2,
                },
                Object {
                  "boardPosition": Object {
                    "x": 0,
                    "y": 1,
                  },
                  "position": Object {
                    "x": 1,
                    "y": 0,
                  },
                  "value": 2,
                },
                Object {
                  "boardPosition": Object {
                    "x": 0,
                    "y": 1,
                  },
                  "position": Object {
                    "x": 1,
                    "y": 1,
                  },
                  "value": 1,
                },
                Object {
                  "boardPosition": Object {
                    "x": 0,
                    "y": 1,
                  },
                  "position": Object {
                    "x": 1,
                    "y": 2,
                  },
                  "value": 2,
                },
                Object {
                  "boardPosition": Object {
                    "x": 0,
                    "y": 1,
                  },
                  "position": Object {
                    "x": 2,
                    "y": 0,
                  },
                  "value": 2,
                },
                Object {
                  "boardPosition": Object {
                    "x": 0,
                    "y": 1,
                  },
                  "position": Object {
                    "x": 2,
                    "y": 1,
                  },
                  "value": 1,
                },
                Object {
                  "boardPosition": Object {
                    "x": 0,
                    "y": 1,
                  },
                  "position": Object {
                    "x": 2,
                    "y": 2,
                  },
                  "value": 2,
                },
              ],
              "value": 1,
            },
            Object {
              "position": Object {
                "x": 0,
                "y": 2,
              },
              "tiles": Array [
                Object {
                  "boardPosition": Object {
                    "x": 0,
                    "y": 2,
                  },
                  "position": Object {
                    "x": 0,
                    "y": 0,
                  },
                  "value": 2,
                },
                Object {
                  "boardPosition": Object {
                    "x": 0,
                    "y": 2,
                  },
                  "position": Object {
                    "x": 0,
                    "y": 1,
                  },
                  "value": 2,
                },
                Object {
                  "boardPosition": Object {
                    "x": 0,
                    "y": 2,
                  },
                  "position": Object {
                    "x": 0,
                    "y": 2,
                  },
                  "value": 1,
                },
                Object {
                  "boardPosition": Object {
                    "x": 0,
                    "y": 2,
                  },
                  "position": Object {
                    "x": 1,
                    "y": 0,
                  },
                  "value": 2,
                },
                Object {
                  "boardPosition": Object {
                    "x": 0,
                    "y": 2,
                  },
                  "position": Object {
                    "x": 1,
                    "y": 1,
                  },
                  "value": 2,
                },
                Object {
                  "boardPosition": Object {
                    "x": 0,
                    "y": 2,
                  },
                  "position": Object {
                    "x": 1,
                    "y": 2,
                  },
                  "value": 1,
                },
                Object {
                  "boardPosition": Object {
                    "x": 0,
                    "y": 2,
                  },
                  "position": Object {
                    "x": 2,
                    "y": 0,
                  },
                  "value": 2,
                },
                Object {
                  "boardPosition": Object {
                    "x": 0,
                    "y": 2,
                  },
                  "position": Object {
                    "x": 2,
                    "y": 1,
                  },
                  "value": 2,
                },
                Object {
                  "boardPosition": Object {
                    "x": 0,
                    "y": 2,
                  },
                  "position": Object {
                    "x": 2,
                    "y": 2,
                  },
                  "value": 1,
                },
              ],
              "value": 1,
            },
            Object {
              "position": Object {
                "x": 1,
                "y": 0,
              },
              "tiles": Array [
                Object {
                  "boardPosition": Object {
                    "x": 1,
                    "y": 0,
                  },
                  "position": Object {
                    "x": 0,
                    "y": 0,
                  },
                  "value": 1,
                },
                Object {
                  "boardPosition": Object {
                    "x": 1,
                    "y": 0,
                  },
                  "position": Object {
                    "x": 0,
                    "y": 1,
                  },
                  "value": 0,
                },
                Object {
                  "boardPosition": Object {
                    "x": 1,
                    "y": 0,
                  },
                  "position": Object {
                    "x": 0,
                    "y": 2,
                  },
                  "value": 2,
                },
                Object {
                  "boardPosition": Object {
                    "x": 1,
                    "y": 0,
                  },
                  "position": Object {
                    "x": 1,
                    "y": 0,
                  },
                  "value": 0,
                },
                Object {
                  "boardPosition": Object {
                    "x": 1,
                    "y": 0,
                  },
                  "position": Object {
                    "x": 1,
                    "y": 1,
                  },
                  "value": 0,
                },
                Object {
                  "boardPosition": Object {
                    "x": 1,
                    "y": 0,
                  },
                  "position": Object {
                    "x": 1,
                    "y": 2,
                  },
                  "value": 1,
                },
                Object {
                  "boardPosition": Object {
                    "x": 1,
                    "y": 0,
                  },
                  "position": Object {
                    "x": 2,
                    "y": 0,
                  },
                  "value": 1,
                },
                Object {
                  "boardPosition": Object {
                    "x": 1,
                    "y": 0,
                  },
                  "position": Object {
                    "x": 2,
                    "y": 1,
                  },
                  "value": 0,
                },
                Object {
                  "boardPosition": Object {
                    "x": 1,
                    "y": 0,
                  },
                  "position": Object {
                    "x": 2,
                    "y": 2,
                  },
                  "value": 2,
                },
              ],
              "value": 0,
            },
            Object {
              "position": Object {
                "x": 1,
                "y": 1,
              },
              "tiles": Array [
                Object {
                  "boardPosition": Object {
                    "x": 1,
                    "y": 1,
                  },
                  "position": Object {
                    "x": 0,
                    "y": 0,
                  },
                  "value": 0,
                },
                Object {
                  "boardPosition": Object {
                    "x": 1,
                    "y": 1,
                  },
                  "position": Object {
                    "x": 0,
                    "y": 1,
                  },
                  "value": 2,
                },
                Object {
                  "boardPosition": Object {
                    "x": 1,
                    "y": 1,
                  },
                  "position": Object {
                    "x": 0,
                    "y": 2,
                  },
                  "value": 2,
                },
                Object {
                  "boardPosition": Object {
                    "x": 1,
                    "y": 1,
                  },
                  "position": Object {
                    "x": 1,
                    "y": 0,
                  },
                  "value": 2,
                },
                Object {
                  "boardPosition": Object {
                    "x": 1,
                    "y": 1,
                  },
                  "position": Object {
                    "x": 1,
                    "y": 1,
                  },
                  "value": 0,
                },
                Object {
                  "boardPosition": Object {
                    "x": 1,
                    "y": 1,
                  },
                  "position": Object {
                    "x": 1,
                    "y": 2,
                  },
                  "value": 2,
                },
                Object {
                  "boardPosition": Object {
                    "x": 1,
                    "y": 1,
                  },
                  "position": Object {
                    "x": 2,
                    "y": 0,
                  },
                  "value": 1,
                },
                Object {
                  "boardPosition": Object {
                    "x": 1,
                    "y": 1,
                  },
                  "position": Object {
                    "x": 2,
                    "y": 1,
                  },
                  "value": 1,
                },
                Object {
                  "boardPosition": Object {
                    "x": 1,
                    "y": 1,
                  },
                  "position": Object {
                    "x": 2,
                    "y": 2,
                  },
                  "value": 1,
                },
              ],
              "value": 1,
            },
            Object {
              "position": Object {
                "x": 1,
                "y": 2,
              },
              "tiles": Array [
                Object {
                  "boardPosition": Object {
                    "x": 1,
                    "y": 2,
                  },
                  "position": Object {
                    "x": 0,
                    "y": 0,
                  },
                  "value": 2,
                },
                Object {
                  "boardPosition": Object {
                    "x": 1,
                    "y": 2,
                  },
                  "position": Object {
                    "x": 0,
                    "y": 1,
                  },
                  "value": 2,
                },
                Object {
                  "boardPosition": Object {
                    "x": 1,
                    "y": 2,
                  },
                  "position": Object {
                    "x": 0,
                    "y": 2,
                  },
                  "value": 0,
                },
                Object {
                  "boardPosition": Object {
                    "x": 1,
                    "y": 2,
                  },
                  "position": Object {
                    "x": 1,
                    "y": 0,
                  },
                  "value": 2,
                },
                Object {
                  "boardPosition": Object {
                    "x": 1,
                    "y": 2,
                  },
                  "position": Object {
                    "x": 1,
                    "y": 1,
                  },
                  "value": 0,
                },
                Object {
                  "boardPosition": Object {
                    "x": 1,
                    "y": 2,
                  },
                  "position": Object {
                    "x": 1,
                    "y": 2,
                  },
                  "value": 2,
                },
                Object {
                  "boardPosition": Object {
                    "x": 1,
                    "y": 2,
                  },
                  "position": Object {
                    "x": 2,
                    "y": 0,
                  },
                  "value": 2,
                },
                Object {
                  "boardPosition": Object {
                    "x": 1,
                    "y": 2,
                  },
                  "position": Object {
                    "x": 2,
                    "y": 1,
                  },
                  "value": 1,
                },
                Object {
                  "boardPosition": Object {
                    "x": 1,
                    "y": 2,
                  },
                  "position": Object {
                    "x": 2,
                    "y": 2,
                  },
                  "value": 2,
                },
              ],
              "value": 2,
            },
            Object {
              "position": Object {
                "x": 2,
                "y": 0,
              },
              "tiles": Array [
                Object {
                  "boardPosition": Object {
                    "x": 2,
                    "y": 0,
                  },
                  "position": Object {
                    "x": 0,
                    "y": 0,
                  },
                  "value": 2,
                },
                Object {
                  "boardPosition": Object {
                    "x": 2,
                    "y": 0,
                  },
                  "position": Object {
                    "x": 0,
                    "y": 1,
                  },
                  "value": 0,
                },
                Object {
                  "boardPosition": Object {
                    "x": 2,
                    "y": 0,
                  },
                  "position": Object {
                    "x": 0,
                    "y": 2,
                  },
                  "value": 2,
                },
                Object {
                  "boardPosition": Object {
                    "x": 2,
                    "y": 0,
                  },
                  "position": Object {
                    "x": 1,
                    "y": 0,
                  },
                  "value": 1,
                },
                Object {
                  "boardPosition": Object {
                    "x": 2,
                    "y": 0,
                  },
                  "position": Object {
                    "x": 1,
                    "y": 1,
                  },
                  "value": 1,
                },
                Object {
                  "boardPosition": Object {
                    "x": 2,
                    "y": 0,
                  },
                  "position": Object {
                    "x": 1,
                    "y": 2,
                  },
                  "value": 2,
                },
                Object {
                  "boardPosition": Object {
                    "x": 2,
                    "y": 0,
                  },
                  "position": Object {
                    "x": 2,
                    "y": 0,
                  },
                  "value": 0,
                },
                Object {
                  "boardPosition": Object {
                    "x": 2,
                    "y": 0,
                  },
                  "position": Object {
                    "x": 2,
                    "y": 1,
                  },
                  "value": 2,
                },
                Object {
                  "boardPosition": Object {
                    "x": 2,
                    "y": 0,
                  },
                  "position": Object {
                    "x": 2,
                    "y": 2,
                  },
                  "value": 2,
                },
              ],
              "value": 2,
            },
            Object {
              "position": Object {
                "x": 2,
                "y": 1,
              },
              "tiles": Array [
                Object {
                  "boardPosition": Object {
                    "x": 2,
                    "y": 1,
                  },
                  "position": Object {
                    "x": 0,
                    "y": 0,
                  },
                  "value": 2,
                },
                Object {
                  "boardPosition": Object {
                    "x": 2,
                    "y": 1,
                  },
                  "position": Object {
                    "x": 0,
                    "y": 1,
                  },
                  "value": 0,
                },
                Object {
                  "boardPosition": Object {
                    "x": 2,
                    "y": 1,
                  },
                  "position": Object {
                    "x": 0,
                    "y": 2,
                  },
                  "value": 2,
                },
                Object {
                  "boardPosition": Object {
                    "x": 2,
                    "y": 1,
                  },
                  "position": Object {
                    "x": 1,
                    "y": 0,
                  },
                  "value": 2,
                },
                Object {
                  "boardPosition": Object {
                    "x": 2,
                    "y": 1,
                  },
                  "position": Object {
                    "x": 1,
                    "y": 1,
                  },
                  "value": 0,
                },
                Object {
                  "boardPosition": Object {
                    "x": 2,
                    "y": 1,
                  },
                  "position": Object {
                    "x": 1,
                    "y": 2,
                  },
                  "value": 0,
                },
                Object {
                  "boardPosition": Object {
                    "x": 2,
                    "y": 1,
                  },
                  "position": Object {
                    "x": 2,
                    "y": 0,
                  },
                  "value": 2,
                },
                Object {
                  "boardPosition": Object {
                    "x": 2,
                    "y": 1,
                  },
                  "position": Object {
                    "x": 2,
                    "y": 1,
                  },
                  "value": 0,
                },
                Object {
                  "boardPosition": Object {
                    "x": 2,
                    "y": 1,
                  },
                  "position": Object {
                    "x": 2,
                    "y": 2,
                  },
                  "value": 2,
                },
              ],
              "value": 0,
            },
            Object {
              "position": Object {
                "x": 2,
                "y": 2,
              },
              "tiles": Array [
                Object {
                  "boardPosition": Object {
                    "x": 2,
                    "y": 2,
                  },
                  "position": Object {
                    "x": 0,
                    "y": 0,
                  },
                  "value": 0,
                },
                Object {
                  "boardPosition": Object {
                    "x": 2,
                    "y": 2,
                  },
                  "position": Object {
                    "x": 0,
                    "y": 1,
                  },
                  "value": 2,
                },
                Object {
                  "boardPosition": Object {
                    "x": 2,
                    "y": 2,
                  },
                  "position": Object {
                    "x": 0,
                    "y": 2,
                  },
                  "value": 0,
                },
                Object {
                  "boardPosition": Object {
                    "x": 2,
                    "y": 2,
                  },
                  "position": Object {
                    "x": 1,
                    "y": 0,
                  },
                  "value": 2,
                },
                Object {
                  "boardPosition": Object {
                    "x": 2,
                    "y": 2,
                  },
                  "position": Object {
                    "x": 1,
                    "y": 1,
                  },
                  "value": 2,
                },
                Object {
                  "boardPosition": Object {
                    "x": 2,
                    "y": 2,
                  },
                  "position": Object {
                    "x": 1,
                    "y": 2,
                  },
                  "value": 2,
                },
                Object {
                  "boardPosition": Object {
                    "x": 2,
                    "y": 2,
                  },
                  "position": Object {
                    "x": 2,
                    "y": 0,
                  },
                  "value": 2,
                },
                Object {
                  "boardPosition": Object {
                    "x": 2,
                    "y": 2,
                  },
                  "position": Object {
                    "x": 2,
                    "y": 1,
                  },
                  "value": 2,
                },
                Object {
                  "boardPosition": Object {
                    "x": 2,
                    "y": 2,
                  },
                  "position": Object {
                    "x": 2,
                    "y": 2,
                  },
                  "value": 2,
                },
              ],
              "value": 2,
            },
          ]
    }
    currentMove={1}
    currentPlayer={0}
    history={[Function]}
    location={[Function]}
    match={
        Object {
            "isExact": true,
            "params": Object {
              "id": "someId",
            },
            "path": "",
            "url": "",
          }
    }
    moveBackwardInHistory={[Function]}
    moveForwardInHistory={[Function]}
    onLoad={[Function]}
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": <div
        className="analysisLayout"
>
        <div
                className="moveList"
                id="moveList"
        />
        <div
                className="historyButtons"
        />
        <div
                className="analysisGame"
        />
</div>,
      "className": "center",
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <div
            className="moveList"
            id="moveList"
/>,
          <div
            className="historyButtons"
/>,
          <div
            className="analysisGame"
/>,
        ],
        "className": "analysisLayout",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": undefined,
            "className": "moveList",
            "id": "moveList",
          },
          "ref": null,
          "rendered": null,
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": undefined,
            "className": "historyButtons",
          },
          "ref": null,
          "rendered": null,
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": undefined,
            "className": "analysisGame",
          },
          "ref": null,
          "rendered": null,
          "type": "div",
        },
      ],
      "type": "div",
    },
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <div
          className="analysisLayout"
>
          <div
                    className="moveList"
                    id="moveList"
          />
          <div
                    className="historyButtons"
          />
          <div
                    className="analysisGame"
          />
</div>,
        "className": "center",
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <div
              className="moveList"
              id="moveList"
/>,
            <div
              className="historyButtons"
/>,
            <div
              className="analysisGame"
/>,
          ],
          "className": "analysisLayout",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": undefined,
              "className": "moveList",
              "id": "moveList",
            },
            "ref": null,
            "rendered": null,
            "type": "div",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": undefined,
              "className": "historyButtons",
            },
            "ref": null,
            "rendered": null,
            "type": "div",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": undefined,
              "className": "analysisGame",
            },
            "ref": null,
            "rendered": null,
            "type": "div",
          },
        ],
        "type": "div",
      },
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromError": true,
          "getDerivedStateFromProps": Object {
            "hasShouldComponentUpdateBug": false,
          },
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
  Symbol(enzyme.__childContext__): null,
}
`;
